Content.Shared/_Moffstation/Cards/Systems/PlayingCardsSystem.Transfer.cs

```diff
@@ -0,0 +1,351 @@
+        entity.Comp switch
+        {
+            PlayingCardDeckComponent deck => AsCardSink((entity, deck)),
+            PlayingCardHandComponent hand => AsCardSink((entity, hand)),
+            _ => throw new($"Unknown variant of {nameof(PlayingCardStackComponent)}"),
+        };
+
+    private CardSink AsCardSink(Entity<PlayingCardDeckComponent> entity) => (cards, user) =>
+    {
+        var deckCoords = Transform(entity).Coordinates;
+
+        Entity<PlayingCardComponent>? first = null;
+        var count = 0;
+        foreach (var cardLike in cards)
+        {
+            count += 1;
+            first ??= cardLike is CardLike.Entity(var ent1) ? ent1 : null;
+            switch (cardLike)
+            {
+                case CardLike.Entity(var ent):
+                    entity.Comp.Cards.Add(new PlayingCardInDeck.NetEnt(GetNetEntity(ent)));
+                    if (!_container.Insert(ent.Owner, entity.Comp.Container, force: true))
+                    {
+                        this.AssertOrLogError(
+                            $"Failed to insert card into deck (card={ToPrettyString(ent)}, hand={ToPrettyString(entity)})"
+                        );
+                    }
+
+                    break;
+                case CardLike.Unspawned(var cardInDeck):
+                    entity.Comp.Cards.Add(cardInDeck);
+                    break;
+                default:
+                    cardLike.ThrowUnknownInheritor();
+                    break;
+            }
+        }
+
+        if (count > 0)
+        {
+            entity.Comp.DirtyVisuals = true;
+            Dirty(entity);
+
+            _audio.PlayPredicted(entity.Comp.PlaceDownSound, deckCoords, user);
+        }
+
+        return (first, count);
+    };
+
+    private CardSink AsCardSink(Entity<PlayingCardHandComponent> entity) => (cards, user) =>
+    {
+        var coords = Transform(entity).Coordinates;
+
+        Entity<PlayingCardComponent>? first = null;
+        var count = 0;
+        foreach (var cardLike in cards)
+        {
+            if (EnsureSpawnedOrNull(cardLike, coords) is not { } spawned)
+                continue;
+            count += 1;
+            first ??= spawned;
+            entity.Comp.Cards.Add(GetNetEntity(spawned));
+            if (!_container.Insert(spawned.Owner, entity.Comp.Container, force: true))
+            {
+                this.AssertOrLogError(
+                    $"Failed to insert card into hand (card={ToPrettyString(spawned)}, hand={ToPrettyString(entity)})"
+                );
+            }
+        }
+
+        if (count > 0)
+        {
+            entity.Comp.DirtyVisuals = true;
+            Dirty(entity);
+
+            _audio.PlayPredicted(entity.Comp.PlaceDownSound, Transform(entity).Coordinates, user);
+        }
+
+        return (first, count);
+    };
+
+    private Entity<PlayingCardComponent>? EnsureSpawnedOrNull(CardLike cardLike, EntityCoordinates coords) =>
```

Comment:
The good news is that making sure a card is spawned is already implemented, just huck it into this function :)